File: sistema_java/codigos/oficinapoo/Paciente.java
package oficinapoo;


public class Paciente extends Pessoa
{
    
    //construtores
    public Paciente()
    {
        super("", "", "12345678900", "33988888888");
    }
    
    public Paciente(String nome, String endereco, String cpf, String telefone)
    {
        super(nome, endereco, cpf, telefone);
    }
    
    public Paciente(Paciente paciente)
    {
        super(paciente.getNome(), paciente.getEndereco(), paciente.getCpf(), paciente.getTelefone());
    }
   

    //getters e setters herdados
	
   //sobrecarga do operador 'printf'
    public String toString()
    {
        return super.toString();
    }

    //sobrecarga do operador ==
    public boolean equals(Object o)
    {
        if(this == o)
        {
            return true;
        }
        
        if(o == null)
        {
            return false;
        }
        
        if(getClass() != o.getClass())
        {
            return false;
        }
        
        final Paciente paciente = (Paciente) o;
        
        return (this.getNome().equals(paciente.getNome()) && this.getEndereco().equals(paciente.getEndereco()) 
		&& this.getCpf().equals(paciente.getCpf()) && this.getTelefone().equals(paciente.getTelefone())); 
    }
}

File: sistema_cpluplus/codigos/Paciente.cpp
#include "Paciente.h"
#include <iostream>//entrada e saida


using namespace std;

//construtor
Paciente::Paciente(string nome, string endereco, long long cpf, long long telefone) : Pessoa(nome, endereco, cpf, telefone)
{
	
}

//construtor
Paciente::Paciente(Paciente& paciente)
{
	setNome(paciente.getNome());
	setEndereco(paciente.getEndereco());
	setCpf(paciente.getCpf());
	setTelefone(paciente.getTelefone());
	
}

//destrutor

Paciente:: ~Paciente() {}


//sobrecarga do operador de atribuicao '=' 

Paciente& Paciente:: operator=(Paciente& paciente)
{
	if (this == &paciente)
	{
		return *this;
	}

	setNome(paciente.getNome());
	setEndereco(paciente.getEndereco());
	setCpf(paciente.getCpf());
	setTelefone(paciente.getTelefone());

	return *this;
}

//sobrecarga do operador 'cout'
ostream& operator << (ostream& out, Paciente& paciente)
{
	out << "Nome: " << paciente.getNome() << "\n" << "Endereco: " << paciente.getEndereco() << "\n" << "Cpf: " << paciente.getCpf()
		<< "\n"
		<<"Telefone: " << paciente.getTelefone();
	return out;
}

//sobrecarga do operador ==
bool Paciente::operator==(Paciente& paciente)
{
	return(this->getNome() == paciente.getNome() && this->getEndereco() == paciente.getEndereco() 
		&& this->getCpf() == paciente.getCpf() && this->getTelefone() == paciente.getTelefone());
}